src/mpi/comm/comm_split_type.c

```diff
@@ -177,6 +182,42 @@ static int node_split_processor(MPIR_Comm * comm_ptr, int key, hwloc_obj_type_t
   fn_fail:
     goto fn_exit;
 }
+
+static int node_split_io(MPIR_Comm * comm_ptr, int key, struct info_obj* info_obj_type,
```

Comment:
`node_split_io` does not seem like a good name since it covers devices that are not I/O devices.